runelite-client/src/main/java/net/runelite/client/plugins/grandexchange/GrandExchangeItemPanel.java

```diff
@@ -127,15 +133,38 @@ public void mouseReleased(MouseEvent e)
 		gePriceLabel.setForeground(ColorScheme.GRAND_EXCHANGE_PRICE);
 		rightPanel.add(gePriceLabel);
 
+		JPanel alchAndLimitPanel = new JPanel(new BorderLayout());
+		panels.add(alchAndLimitPanel);
+		alchAndLimitPanel.setBackground(background);
+
 		// Alch price
 		JLabel haPriceLabel = new JLabel();
 		haPriceLabel.setText(StackFormatter.formatNumber(haPrice.intValue()) + " alch");
 		haPriceLabel.setForeground(ColorScheme.GRAND_EXCHANGE_ALCH);
-		rightPanel.add(haPriceLabel);
+		alchAndLimitPanel.add(haPriceLabel, BorderLayout.WEST);
+
+		// GE Limit
+		JLabel geLimitLabel = new JLabel();
+		String limitLabelText = geItemLimit == 0 ? "" : "Limit " + StackFormatter.formatNumber(geItemLimit);
+		geLimitLabel.setText(limitLabelText);
+		geLimitLabel.setForeground(ColorScheme.GRAND_EXCHANGE_LIMIT);
+		geLimitLabel.setBorder(new CompoundBorder(geLimitLabel.getBorder(), new EmptyBorder(0, 0, 0, 7)));
+		alchAndLimitPanel.add(geLimitLabel, BorderLayout.EAST);
+
+		rightPanel.add(alchAndLimitPanel);
 
 		add(rightPanel, BorderLayout.CENTER);
 	}
 
+	private void matchComponentBackground(JPanel panel, Color color)
```

Comment:
Can't you just override the setBackground method?

Reply:
And inside it, call the super.setBackground, along with that for loop, to apply the color to its children

Reply:
I agree, I believe that would be cleaner.

Reply:
I've looked into it again, the reason why I did it this way was that the loop of children only looks down a single level. All the children are considered components. This means I cannot recursively look for more components from each child since that method does not exist within the component class. Casting everything to a JPanel also gave me runtime erros for certain classes.